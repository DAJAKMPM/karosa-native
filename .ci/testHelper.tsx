import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
