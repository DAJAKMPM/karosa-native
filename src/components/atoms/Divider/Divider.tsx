/**
 *
 * Divider
 * @format
 *
 */

import React, { FC } from "react";
import { Divider as RnDivider } from "react-native-elements";

import type { PropsType } from "./types";

const Divider: FC<PropsType> = (props) => {
  const { style } = props;

  return <RnDivider style={style} />;
};

export default Divider;
