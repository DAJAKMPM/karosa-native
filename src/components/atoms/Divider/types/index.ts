/**
 *
 * @format
 *
 */

import { StyleProp, ViewStyle } from "react-native";

export type PropsType = {
  style?: StyleProp<ViewStyle>;
};
