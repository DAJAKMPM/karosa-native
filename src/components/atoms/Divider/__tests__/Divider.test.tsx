import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import Divider from "../Divider";

describe("Divider Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(<Divider />);

    expect(toJSON()).toMatchSnapshot();
  });
});
