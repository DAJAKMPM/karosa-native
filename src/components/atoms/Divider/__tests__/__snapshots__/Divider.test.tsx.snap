// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      Object {
        "borderBottomColor": "#bcbbc1",
        "borderBottomWidth": 0.5,
      }
    }
  />
</View>
`;
